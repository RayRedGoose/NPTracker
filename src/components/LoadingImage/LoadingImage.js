import './LoadingImage.scss'
import React from 'react'
import balalaika from 'assets/balalaika.svg'
import note from 'assets/note.svg'
import noteOne from 'assets/note-1.svg'
import noteTwo from 'assets/note-2.svg'
import noteThree from 'assets/note-3.svg'

const LoadingImage = () => {
  return (
    <section className="loading-image">
      <div>
        <img src={note} alt="note"/>
        <img src={noteOne} alt="note"/>
        <img src={noteTwo} alt="note"/>
        <img src={noteThree} alt="note"/>
      </div>
      <img src={balalaika} alt="balalaika"/>
      <p>loading...</p>
    </section>
  )
}

export default LoadingImage
