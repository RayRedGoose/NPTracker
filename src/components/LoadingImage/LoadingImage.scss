@import "../../variables.scss";

.loading-image {
  display: flex;
  justify-content: center;
  width: 100%;
  p {
    background-color: $bg-color-primary;
    color: $accent-color-primary-dark;
    font-weight: 700;
    letter-spacing: 0.5em;
    position: absolute;
    bottom: 30vh;
    padding: 10px;
  }
  &>img {
    width: 150px;
  }
  div {
    img {
      width: 30px;
      position: absolute;
      top: 50vh;
      $total: 4;
      @for $i from 1 through $total {
        &:nth-child(#{$i}) {
          $new: $i - 1;
          transform: translate($new * 25px, -$new * 10px);
          animation: fly-#{$i} 5s $new*3s linear infinite;
          opacity: 0;
        }

        @keyframes fly-#{$i} {
          0% {
            opacity: 0;
          }
          20% {
            opacity: 1;
          }
          90% {
            opacity: 1;
          }
          100% {
            $new: 1 - $i;
            transform: translate(-$new*2vw, -20vh);
            opacity: 0;
          }
        }
      }
    }
  }
}
