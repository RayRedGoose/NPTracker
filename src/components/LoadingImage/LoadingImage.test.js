import React from 'react'
import { shallow } from 'enzyme'
import LoadingImage from './LoadingImage'

describe('LoadingImage', () => {
  it('should match the snapshot with all data passed in correctly', () => {
    const block = shallow(<LoadingImage />)

    expect(block).toMatchSnapshot()
  })
})
