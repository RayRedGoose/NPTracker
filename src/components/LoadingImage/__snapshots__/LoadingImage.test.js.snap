// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoadingImage should match the snapshot with all data passed in correctly 1`] = `
<section
  className="loading-image"
>
  <div>
    <img
      alt="note"
      src="note.svg"
    />
    <img
      alt="note"
      src="note-1.svg"
    />
    <img
      alt="note"
      src="note-2.svg"
    />
    <img
      alt="note"
      src="note-3.svg"
    />
  </div>
  <img
    alt="balalaika"
    src="balalaika.svg"
  />
  <p>
    loading...
  </p>
</section>
`;
